### PR TITLE
Unit tests governance smart contract

### PR DESCRIPTION
**Pull request by miiu96**
Added a lot of unit tests for governance system smart contract also did some fixes in the contract.a

### CODE REVIEW

File: vm/systemSmartContracts/governance_test.go [hunk 1]
```diff
@@ -1 +1,448 @@
...
+	callInput := createVMInput(big.NewInt(0), "vote", callerAddr, []byte("addr2"))
+	callInput.Arguments = [][]byte{
+		proposalToVote,
+		vote,
+		validatorAddr,
+	}
+
+	retCode := gsc.Execute(callInput)
+	require.Equal(t, vmcommon.Ok, retCode)
+}
```
Comment (sasurobert) [inline vm/systemSmartContracts/governance_test.go:448]: we need another test -> create a general proposal - add multiple voting transactions and one transaction to finalize it. Check in the end the ProposalStatus. This should be done without mocks. For example look in test: TestStakingAuctionSC_ExecuteStakeUnStakeOneBlsPubKeyAndRestake

File: vm/systemSmartContracts/governance.go [hunk 2]
```diff
@@ -124,8 +125,7 @@ func (g *governanceContract) init(args *vmcommon.ContractCallInput) vmcommon.Ret
 
 	g.eei.SetStorage([]byte(governanceConfigKey), marshaledData)
 	g.eei.SetStorage([]byte(ownerKey), args.CallerAddr)
-	g.ownerAddress = make([]byte, 0, len(args.CallerAddr))
-	g.ownerAddress = append(g.ownerAddress, args.CallerAddr...)
+	g.ownerAddress = args.CallerAddr
```
[the remark this answers is not in the document]
Comment (miiu96) [reply vm/systemSmartContracts/governance.go:128]: done.

File: vm/systemSmartContracts/governance_test.go (hunk 1 above)
Comment (miiu96) [reply vm/systemSmartContracts/governance_test.go:454]: done.

File: vm/systemSmartContracts/governance_test.go [hunk 3]
```diff
@@ -1 +1,614 @@
...
+
+// Test Scenario
+// A proposal is voted if it has 2 vote with yes
+// 1. Init governance smart contract
+// 2. WhiteList an address at genesis
+// 3. WhileList a new address, vote with 2 validators with yes and close white list proposal
+// 4. Create a new general proposal with new white listed address, vote with 2 validator with yes general proposal
+// and close proposal
+func TestGovernanceContract_ExecuteProposalCloseProposal(t *testing.T) {
```
Comment (sasurobert) [inline vm/systemSmartContracts/governance_test.go:463]: you might add a function for each step 1, 2, 3 and 4

File: vm/mock/auctionSCMock.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package mock
+
+import vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+
+// AuctionSCMock -
+type AuctionSCMock struct {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vm/mock/auctionSCMock.go:6]: and you already have SystemSCStub in the mock folder

[the remark this answers is not in the document]
Comment (miiu96) [reply vm/mock/auctionSCMock.go:6]: done.

File: vm/systemSmartContracts/governance_test.go (hunk 3 above)
Comment (miiu96) [reply vm/systemSmartContracts/governance_test.go:463]: done.

**Pull request merged**